frontend/src/components/query/query-builder/query-builder.tsx

```diff
@@ -115,10 +119,35 @@ export default function QueryBuilder({
           />
         );
       case StepIdentifierEnum.Relate:
+        return (
+          <RelateStepComponent
+            key={create ? "new step" : index}
+            index={index}
+            step={!!step ? (step as RelateStep) : step}
```

Comment:
Do we need !!? step will evaluate as a condition without the conversion to boolean